components/calendar/src/astronomy.rs

```diff
@@ -1165,32 +1165,35 @@ impl Astronomical {
     ///
     /// Based on functions from _Calendrical Calculations_ by Reingold & Dershowitz.
     /// Reference lisp code: https://github.com/EdReingold/calendar-code2/blob/9afc1f3/calendar.l#L6883-L6896
-    pub fn phasis_on_or_after(date: RataDie, location: Location) -> RataDie {
-        let moon = libm::floor(Self::lunar_phase_at_or_before(0.0, date.as_moment()).inner());
-        let age = date.to_f64_date() - moon;
+    pub fn phasis_on_or_after(date: RataDie, location: Location, lunar_phase: f64) -> RataDie {
```

Comment:
Suggestion: For this function it seems better to make it an Option which you compute if `None` since it is derivable with the other two arguments.

```suggestion
    pub fn phasis_on_or_after(date: RataDie, location: Location, lunar_phase: Option<f64>) -> RataDie {
```